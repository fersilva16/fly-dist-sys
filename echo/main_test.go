package main

import (
	"encoding/json"
	"fly-dist-sys/testutils"
	"io"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	maelstrom "github.com/jepsen-io/maelstrom/demo/go"
	"github.com/stretchr/testify/require"
)

func TestEcho(t *testing.T) {
	require := require.New(t)

	var stdin io.WriteCloser
	var stdout io.ReadCloser

	node, stdin, stdout = testutils.NewNode()

	go main()

	init_err := testutils.InitNode(stdin, stdout, "n0", []string{"n0"})

	require.NoError(init_err)

	body, body_err := json.Marshal(EchoRequest{
		MessageBody: maelstrom.MessageBody{
			Type:  "echo",
			MsgID: 2,
		},

		Echo: "Please echo 1",
	})

	require.NoError(body_err)

	send_err := testutils.Send(stdin, body)

	require.NoError(send_err)

	output, read_err := testutils.Read(stdout)

	require.NoError(read_err)

	snaps.MatchSnapshot(t, output)
}
