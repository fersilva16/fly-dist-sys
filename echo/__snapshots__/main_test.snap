
[TestEcho - 1]
{"src":"n0","dest":"c0","body":{"echo":"Please echo 1","in_reply_to":2,"type":"echo_ok"}}
---
